packages/angular_devkit/build_facade/src/target.ts

```diff
@@ -0,0 +1,81 @@
+/**
+ * @license
+ * Copyright Google Inc. All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+import {
+  BaseException,
+  Path,
+  dirname,
+  getSystemPath,
+  logging,
+  schema,
+  virtualFs,
+} from '@angular-devkit/core';
+import * as fs from 'fs';
+import { Observable } from 'rxjs/Observable';
+import { of } from 'rxjs/observable/of';
+import { _throw } from 'rxjs/observable/throw';
+import { mergeMap } from 'rxjs/operators';
+import { Architect, TargetInfo } from './architect';
+
+
+export class RunImplementationNotFoundException extends BaseException {
+  constructor() {
+    super(`Target must provide 'run' method implementation.`);
+  }
+}
+
+export class SchemaValidationException extends BaseException {
+  constructor(errors: string[]) {
+    super(`Schema validation failed with the following errors:\n  ${errors.join('\n  ')}`);
+  }
+}
+
+export interface TargetContext {
+  logger: logging.Logger;
+  host: virtualFs.SyncDelegateHost<fs.Stats>;
+  architect: Architect;
+}
+
+// TODO: use Build Event Protocol
+// https://docs.bazel.build/versions/master/build-event-protocol.html
+// https://github.com/googleapis/googleapis/tree/master/google/devtools/build/v1
+export interface BuildEvent {
+  success: boolean;
+}
+
+// TODO: need a way to show/return option help.
+export class Target<OptionsT> {
+  constructor(public schemaPath: Path, public context: TargetContext) { }
+
+  protected validateOptions(options: Partial<OptionsT>): Observable<OptionsT> {
+    const schemaContent = virtualFs.fileBufferToString(this.context.host.read(this.schemaPath));
```

Comment:
We need to use host everywhere. SyncHost isn't compatible with a bunch of use cases.

Reply:
Done.